Caltech-IPAC/firefly
# Firefly-1503: Update eslint, node and other dependencies

Fixes [FIREFLY-1503](https://jira.ipac.caltech.edu/browse/FIREFLY-1503)

- Updated eslint and related dependencies (eslint plugins)
- Replaced `.eslintrc` by `eslint.config.mjs` using Eslint 9 [configuration migrator](https://eslint.org/docs/latest/use/configure/migration-guide#migrate-your-config-file). Additional modifications:
  - Made it work with both js and jsx
  - Ignore gitignored files
  - Fixed deprecated rules (now available through `@stylistic/js` pacakge)
  - Reordered and regrouped rules based on plugins
- Upgraded node version in Dockerfile and readme
- Upgraded dependencies with vulnerability warnings: webpack, @babel/core - also updated related packages of babel
- Updated documentation to update package.json: https://github.com/IPAC-SW/irsa-ife/pull/338

## Testing
- Build should be successful: https://irsajenkins.ipac.caltech.edu/job/firefly/271/console
- Checkout this branch locally, and go to intellij and check `Firefly.js` and `TapViewType.jsx` files have eslint warnings. Can also use `yarn run eslint src/firefly/js/ui/tap/TapViewType.jsx --debug` if testing through CLI.
- No impact on UI: https://fireflydev.ipac.caltech.edu/firefly-1503-eslint-update/firefly/
